lte/gateway/c/core/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -2278,21 +2329,44 @@ imsi64_t mme_app_handle_initial_paging_request(
     OAILOG_FUNC_RETURN(LOG_MME_APP, RETURNok);
   }
   imsi64_t* imsi_list = NULL;
-  OAILOG_DEBUG(
-      LOG_MME_APP, "paging is requested for ue_ip:%x \n",
-      paging_req->ipv4_addr.s_addr);
-  int num_imsis =
-      mme_app_get_imsi_from_ipv4(paging_req->ipv4_addr.s_addr, &imsi_list);
-  if (!(num_imsis)) {
-    OAILOG_ERROR(
-        LOG_MME_APP, "Failed to fetch imsi from ue_ip:%x \n",
-        paging_req->ipv4_addr.s_addr);
-    OAILOG_FUNC_RETURN(LOG_MME_APP, RETURNerror);
-  }
-  for (int idx = 0; idx < num_imsis; idx++) {
-    if (imsi_list[idx] != INVALID_IMSI64) {
-      imsi64 = imsi_list[idx];
-      mme_app_send_paging_request(mme_app_desc_p, imsi64);
+
+  if (paging_req->ip_addr_type == IPV4_ADDR_TYPE) {
+    OAILOG_DEBUG_UE(
+        LOG_MME_APP, imsi64, "paging is requested for ue_ipv4:%x \n",
+        paging_req->address.ipv4_addr.sin_addr);
+    int num_imsis = mme_app_get_imsi_from_ipv4(
+        paging_req->address.ipv4_addr.sin_addr.s_addr, &imsi_list);
+    if (!(num_imsis)) {
+      OAILOG_ERROR_UE(
+          LOG_MME_APP, imsi64, "Failed to fetch imsi from ue_ipv4:%x \n",
+          paging_req->address.ipv4_addr);
```

Comment:
```suggestion
          paging_req->address.ipv4_addr.s_addr);
```
I think we're missing passing the `s_addr`?